localstack/utils/venv.py

```diff
@@ -0,0 +1,100 @@
+import io
+import os
+import sys
+from functools import cached_property
+from pathlib import Path
+from typing import Union
+
+
+class VirtualEnvironment:
+    """
+    Encapsulates methods to operate and navigate on a python virtual environment.
+    """
+
+    def __init__(self, venv_dir: str | os.PathLike[str]):
+        self._venv_dir = venv_dir
+
+    def create(self):
+        """
+        Uses the virtualenv cli to create the virtual environment.
+        :return:
+        """
+        self.venv_dir.mkdir(parents=True, exist_ok=True)
+        from venv import main
+
+        main([str(self.venv_dir)])
+
+    @property
+    def exists(self) -> bool:
+        """
+        Checks whether the virtual environment exists by checking whether the site-package directory of the venv exists.
+        :return: the if the venv exists
+        :raises NotADirectoryError: if the venv path exists but is not a directory
+        """
+        try:
+            return True if self.site_dir else False
+        except FileNotFoundError:
```

Comment:
nit: Not sure if we also want to catch `NotADirectoryError` here - although in that case it seems fine to bubble up the exception, as it will likely require manual fixes by the user.. 👍

Reply:
when the site_dir path exists but is not a directory, then it's not clear to me what the semantics of `exists` is anymore. if it's not a directory someone has probably manually tampered with something and they deserve the exception :P personally i think it's OK if this system exception bubbles upwards, because it's similar to an unexpected IOError.